components/Starknet/modules/tools/pages/starkgate_dev_reference.adoc

```diff
@@ -0,0 +1,490 @@
+
+None
+
+[discrete]
+==== Returns
+
+The address of the Registry contract.
+
+[discrete]
+==== Function definition
+
+link:https://github.com/starkware-libs/starkgate/blob/7b967dd0db1019e91b1efb5e530fdef9f025e538/src/solidity/IStarkgateManager.sol#L8C14-L8C14[getRegistry]
+
+
+[#isServicingToken]
+=== `isServicingToken`
+
+[discrete]
+==== Description
+
+Checks whether the calling contract is currently providing a service for the specified token.
+
+If `true`, or if `isServicingToken` is not implemented, this function fails and the entire transaction is reverted.
+
+You need to include a function in your bridge that implements `isServicingToken` to check if the bridge is currently servicing any transactions.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="20"]
+`token`:: The address of the contract for the desired token.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="20"]
+`true`:: The calling contract is currently providing a service for the token.
+`false:: The calling contract is not currently providing a service for the token.
+
+[discrete]
+==== Example
+
+In this example, the function `onlyServicingToken` implements the `isServicingToken` API to check if the specified token is currently being serviced. If `isServicingToken` returns `false`,  it outputs `TOKEN_NOT_SERVICED`.
+
+[source,solidity]
+----
+    modifier onlyServicingToken(address token) {
+        require(isServicingToken(token), "TOKEN_NOT_SERVICED");
+        _;
+    }
+----
+
+[discrete]
+==== Function definition
+
+link:https://github.com/starkware-libs/starkgate/blob/f060fd5d1ca49a0014dbcb0777a0c084c12f41d2/src/solidity/IStarkgateService.sol#L9C14-L9C30[isServicingToken] in `IStarkgateService.sol`.
+
+
+
+[#selfRemove]
+=== `selfRemove`
+
+[discrete]
+==== Description
+
+Use this function to remove enlisting of a specific token bridge from the registry.
+
+The bridge must implement the `isServicingToken` function. For more information, see xref:#isServicingToken[`isServicingToken`]).
+
+If you don’t implement `isServicingToken`, or if it returns `true`, this function fails and the entire transaction is reverted.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="20"]
+`token`:: The address of the token bridge contract to remove.
+
+[discrete]
+==== Returns
+
+None.
+
+[discrete]
+==== Function definition
+
+link:https://github.com/starkware-libs/starkgate/blob/f060fd5d1ca49a0014dbcb0777a0c084c12f41d2/src/solidity/IStarkgateRegistry.sol#L37[`selfRemove`] in `IStarkgateRegistry.sol`
+
+
+
+== L2 function reference
+
+
+[#get_erc20_class_hash]
+=== `get_erc20_class_hash`
+
+[discrete]
+==== Description
+
+Returns the class hash of the implementation used by the ERC-20 contract. Use the class hash as the type when deploying the ERC-20 contract on L2.
+
+Creating a new bridge with xref:#enrollTokenBridge[`enrollTokenBridge`] generates this class hash.
```

Comment:
this sounds weird

Reply:
the class hash is the has of the compiled contract file, calling a function on a smart contract can't generate a class hash. At most it can return it but i'm not sure what's the correct term here as idk what `enrollTokenBridge` is doing